src/components/ProFullCourt/index.tsx

```diff
@@ -58,7 +62,14 @@ const ProFullCourt = () => {
     return () => window.removeEventListener("resize", checkSize);
   }, []);
 
-  tileCalculation(canvasRef, courtAndTileInfo);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      const tileNumberResult = tileCalculation(canvasRef, courtAndTileInfo);
+      dispatch(changeTileQuantity(tileNumberResult));
+    }, 100);
+    return () => clearTimeout(timer);
+  }, [tileColorState]);

```

Comment:
I dont get why you run this calculation and dispatch after 100ms. Please explain.

Reply:
If I remove setTimeout (you can try to comment them out),  and run this calculation and dispatch at the same time tileColorState changed, the data that are being calculated and dispatched are still the state before the color change. There will be a bug that result shown in price bar will be the previous result. setTimeout is the solution for this bug.

Reply:
Yes, you are right. Seems like this calculation needs time to send the result back so the whole calculation and dispatch need to be async instead of sync. Even using "0" as the second parameter in setTimeout will do the trick as well. I will look deep into this function tonight or tomorrow to see if there is any proper way to fix this issue. Otherwise, nice work. 👍